src/effect.cpp

```diff
@@ -422,153 +593,114 @@ struct desc_freq {
     desc_freq(double c, int v, std::string pos, std::string neg) : chance(c), val(v), pos_string(pos), neg_string(neg) {};
 };
 
-std::string effect::disp_desc(bool reduced) const
+std::string effect::disp_desc(bool const reduced) const
 {
     std::stringstream ret;
-    // First print stat changes, adding + if value is positive
-    int tmp = get_avg_mod("STR", reduced);
-    if (tmp > 0) {
-        ret << string_format(_("Strength +%d;  "), tmp);
-    } else if (tmp < 0) {
-        ret << string_format(_("Strength %d;  "), tmp);
-    }
-    tmp = get_avg_mod("DEX", reduced);
-    if (tmp > 0) {
-        ret << string_format(_("Dexterity +%d;  "), tmp);
-    } else if (tmp < 0) {
-        ret << string_format(_("Dexterity %d;  "), tmp);
-    }
-    tmp = get_avg_mod("PER", reduced);
-    if (tmp > 0) {
-        ret << string_format(_("Perception +%d;  "), tmp);
-    } else if (tmp < 0) {
-        ret << string_format(_("Perception %d;  "), tmp);
-    }
-    tmp = get_avg_mod("INT", reduced);
-    if (tmp > 0) {
-        ret << string_format(_("Intelligence +%d;  "), tmp);
-    } else if (tmp < 0) {
-        ret << string_format(_("Intelligence %d;  "), tmp);
-    }
-    tmp = get_avg_mod("SPEED", reduced);
-    if (tmp > 0) {
-        ret << string_format(_("Speed +%d;  "), tmp);
-    } else if (tmp < 0) {
-        ret << string_format(_("Speed %d;  "), tmp);
-    }
-    // Newline if necessary
-    if (ret.str() != "") {
-        ret << "\n";
-    }
+
+    auto const write_stat = [&](std::string const& stat, char const* fmt) {
+        auto const value = get_avg_mod(stat, reduced);
+        
+        if (value != 0) {
+            ret << string_format(_(fmt), value);
+        }
+    };
+
+    auto const write_stats = [&] {
```

Comment:
This is a relic of when I was factoring out common chunks of code. It's fixed in my branch.
Too much lambda is noted though.